fs.go

```diff
@@ -50,37 +50,41 @@ func (s *System) GetAllFileSystems() ([]types.FileSystem, error) {
 	return fs, nil
 }
 
-// GetFileSystemByID returns a file system by ID
-func (s *System) GetFileSystemByID(id string) (*types.FileSystem, error) {
+// GetFileSystemByIDName returns a file system by Name or ID
+func (s *System) GetFileSystemByIDName(id string, name string) (*types.FileSystem, error) {
 	defer TimeSpent("GetFileSystemByID", time.Now())
 
-	path := fmt.Sprintf("/rest/v1/file-systems/%v?select=*", id)
-	var fs types.FileSystem
-	err := s.client.getJSONWithRetry(
-		http.MethodGet, path, nil, &fs)
-	if err != nil {
-		return nil, err
-	}
+	if id == "" && name == "" {
 
-	return &fs, nil
-}
+		return nil, errors.New("file system name or ID is mandatory for fetching file system details, please enter a valid value")
+
+	} else if id != "" {
```

Comment:
Since you are returning the value this else if and other else are not required. you can give it like
if id!=""{
  return //something
}

if name != ""{
   return //something
}